V2.17-RUN-16823-whats-new-only (#738)
run-ai/docs: docs/home/whats-new-2-17.md

    * Throughput
    * Concurrency
  
  For more information, see [Autoscaling metrics](../admin/workloads/inference-overview.md#autoscaling).
  
* <!-- RUN-11488/RUN-17720 deprecation of direct metrics in favor of API -->Improved functionality of metrics by using Run:ai APIs. Using the API endpoints is more efficient and provides an easier way of retrieving metrics in any application. For more information, see [Metrics](../developer/metrics/metrics.md#metrics-apis).

#### Authentication and Authorization

* <!-- RUN-15585/RUN-15586 -  View & Edit SSO settings - SAML -->Added new functionality to SAML 2 SSO configuration to assist with troubleshooting login or permissions issues that may arise. Now administrators have the ability to:

    * View and edit SSO settings.
    * Upload or download a settings XML file.

For more information, see [SSO UI configuration](../admin/runai-setup/authentication/sso.md#step-1-ui-configuration).

#### Policies

### Control and Visibility

## Deprecation Notifications

> Deprecation notifications allow you to plan for future changes in the Run:ai Platform. Deprecated features will be available for **two** versions ahead of the notification. For questions, see your Run:ai representative.

Reviewer: @jasonnovichRunAI we need to make sure the "two" doesn't refer to the APIs part.... maybe we shouldn't mention a timeframe in this sentence?

### Feature deprecations

> The following features have been marked for deprecation:

Reviewer: @jasonnovichRunAI will this be removed if there are no deprecated features?

Author: Sort of, if there are no feature deprecatoins, we can either remove it or rewrite the sentnece to say there are none at this time. But if I am not mistaken the "Jobs" and "Deployments" features are deprecated.

#### Authentication and Authorization

* <!-- RUN-15585/RUN-15586 -  View & Edit SSO settings - SAML -->Added new functionality to SAML 2 in the *SSO configuration* in the *Security* category* of the *Tools & Settings* menu. The added functionality assists with troubleshooting login or permissions issues that may arise. Now administrators now have the ability to:

    * View and edit SSO settings.
    * Upload or download a settings XML file.

For more information, see [SSO UI configuration](../admin/runai-setup/authentication/sso.md#step-1-ui-configuration).

<!-- #### Policies - 
no new stories here

### Control and Visibility

no new stories here -->

## Deprecation Notifications

Deprecation notifications allow you to plan for future changes in the Run:ai Platform.

### Feature deprecations

Deprecated features will be available for **two** versions ahead of the notification. For questions, see your Run:ai representative. The following features have been marked for deprecation:

### API support and endpoint deprecations

> The endpoints and parameters specified in the API reference are the ones that are officially supported by Run:ai. For more information about Run:ai's API support policy, see [Developer overview](../developer/overview-developer.md#overview-developer-documentation).

Reviewer: ```suggestion
The endpoints and parameters specified in the API reference are the ones that are officially supported by Run:ai. For more information about Run:ai's API support policy and deprecation process, see [Developer overview](../developer/overview-developer.md#overview-developer-documentation).
```

The following list of API endpoints have been marked for deprecation:

> * Jobs&mdash;the *Jobs* feature (submission form and view) has been moved to the category of *Legacy*. To enable them, go to *Tools & Settings*, *General*, open the *Workloads* pane, and then toggle the *Jobs view* and *Job submission* switch to the enabled position.

Reviewer: @jasonnovichRunAI it seems like this section should be under deprecated features and not deprecated API endpoints, isn't it?

* Deployments&mdash;the *Deployments* feature has been removed. It has been replaced by *Inference* workloads. For more information, see [Jobs, Workloads, and Workspaces](#jobs-workloads-and-workspaces) above.
* Workspaces view&mdash;the *Workspaces* menu has been removed. You can now submit a *Workspace* workload using the *+ New workload* form from the *Workloads* table.

#### Jobs, events, pods API (replaced by workloads/pods/events)

| Deprecated endpoint | Replacement endpoint |
| -- | -- |
| https://app.run.ai/v1/k8s/clusters/{uuid}/jobs | https://app.run.ai/api/v1/workloads |
| https://app.run.ai/v1/k8s/clusters/{uuid}/jobs/count | https://app.run.ai/api/v1/workloads/count |
| https://app.run.ai/v1/k8s/clusters/{uuid}/jobs/{jobId}/pods | https://app.run.ai/api/v1/workloads/{workloadId}/pods |
| https://app.run.ai/v1/k8s/clusters/{uuid}/pods | https://app.run.ai/api/v1/workloads/pods | 

> 

Reviewer: Need to add additional APIs 

```suggestion
### Users, Applications, and Groups API 

| Deprecated endpoint | Replacement endpoint |
| -- | -- |
| https://app.run.ai/v1/k8s/apps | https://app.run.ai/api/v1/k8s/apps |
| https://app.run.ai/v1/k8s/users | https://app.run.ai/api/v1/k8s/users |
| https://app.run.ai/v1/k8s/groups | https://app.run.ai/api/v1/authorization/access-rules (groups should no longer be created, you can only add access rules to them) |
```

> #### cluster metrics (v1) - replaced v2

Reviewer: ```suggestion
#### Cluster metrics v1 (replaced by v2)
```

| Deprecated endpoint | Replacement endpoint |
| -- | -- |
| https://app.run.ai/v1/k8s/clusters/{clusterUuid}/metrics | https://app.run.ai/api/v2/clusters/{clusterUuid}/metrics |

> In some cases name of metrics have been changed. For more information, see [Metrics](../developer/metrics/metrics.md#changed-metrics-and-api-mapping).

Reviewer 2: Should be in a section of it's own. It's important that metrics users will understand that starting cluster version 2.17 some names have changed. It's a breakage they need to be aware of.

Reviewer: @jasonnovichRunAI

| https://app.run.ai/v1/k8s/clusters/{uuid}/jobs/count | https://app.run.ai/api/v1/workloads/count |
| https://app.run.ai/v1/k8s/clusters/{uuid}/jobs/{jobId}/pods | https://app.run.ai/api/v1/workloads/{workloadId}/pods |
| https://app.run.ai/v1/k8s/clusters/{uuid}/pods | https://app.run.ai/api/v1/workloads/pods |

### Users, Applications, and Groups API

| Deprecated endpoint | Replacement endpoint |
| -- | -- |
| https://app.run.ai/v1/k8s/apps | https://app.run.ai/api/v1/k8s/apps |
| https://app.run.ai/v1/k8s/users | https://app.run.ai/api/v1/k8s/users |
| https://app.run.ai/v1/k8s/groups | https://app.run.ai/api/v1/authorization/access-rules (groups should no longer be created, you can only add access rules to them) |

#### Cluster metrics v1 (replaced by v2)

| Deprecated endpoint | Replacement endpoint |
| -- | -- |
| https://app.run.ai/v1/k8s/clusters/{clusterUuid}/metrics | https://app.run.ai/api/v2/clusters/{clusterUuid}/metrics |

## Breaking changes

Breaking changes notifications allow you to plan around potential changes that may interfere your current workflow when interfacing with the Run:ai Platform.

### Metrics

> Be aware that some names of metrics have been changed and/or have been replaced by the Metrics API. For more information, see [Metrics](../developer/metrics/metrics.md#changed-metrics-and-api-mapping).

Reviewer: @lavianalon is it correct to say "**replaced** by the Metrics API" or should we say that it was added?
Maybe it is also worth mentioning that we recommend start using the APIs instead of fetching the metrics directly, WDYT?

Reviewer 2: I think we should say "added API as the preferred way to consume" or something of that sort.
It should be clear that the metrics are still available in the old form in 2.17

* <!-- RUN-15878/RUN-16796/RUN-15878/RUN-16085 omit Knative metrics and update supported metrics and hpa autoscaler -->Updated the knative and autoscaler metrics. Run:ai currently supports the following metrics:

    * Throughput
    * Concurrency
  
    For more information, see [Autoscaling metrics](../admin/workloads/inference-overview.md#autoscaling).
  
* <!-- RUN-11488/RUN-17720 deprecation of direct metrics in favor of API -->Improved availability of metrics by using Run:ai APIs. Using the API endpoints is now the preferred method to retrieve metrics for use in any application. For more information, see [Metrics](../developer/metrics/metrics.md#changed-metrics-and-api-mapping).

#### Authentication and Authorization

> * <!-- RUN-15585/RUN-15586 -  View & Edit SSO settings - SAML -->Added new functionality to SAML 2 in the *SSO configuration* in the *Security* category* of the *Tools & Settings* menu. The added functionality assists with troubleshooting login or permissions issues that may arise. Now administrators now have the ability to:

Reviewer: @jasonnovichRunAI 
```suggestion
* <!-- RUN-15585/RUN-15586 -  View & Edit SSO settings - SAML -->Added new functionality to SAML 2.0 identity provider configuration in the *Security* category of the *General* settings. The added functionality assists with troubleshooting SSO configuration and authentication issues that may arise. Now administrators now have the ability to:
```

>     * View and edit SSO settings.

Reviewer: ```suggestion
    * View and edit the identity provider settings for SAML 2.0
```

>     * Upload or download a settings XML file.

Reviewer: ```suggestion
    * Upload or download the SAML 2.0 identity provider metadata XML file.
```
